### PR TITLE
Feature/applying user

### PR DESCRIPTION
**Pull request by Eukst**
In This PR , Added Following Features
1. User can See the available jobs
2. employer can post , see , edit, delete any job
3. employer can see which user applied for posted job 
4. Home page will change according to user role

### CODE REVIEW

File: app/controllers/apply_jobs_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+class ApplyJobsController < ApplicationController
+  def show
+    if user_signed_in? && (current_user.user_role.include? 'Job Seeker')
+      job_id = params[:id].to_i
+      user_id = current_user.id
+      @apply_job = ApplyJob.create(user_id: user_id, job_id: job_id)
+      if @apply_job.valid?
+        @apply_job.save
+        flash[:success] = 'Applied to Job  Successfully '
+
```
Comment (Sharee-ongraph) [inline app/controllers/apply_jobs_controller.rb]: remove extra spaces

File: app/controllers/jobs_controller.rb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+  # GET /jobs or /jobs.json
+  def index
+    @jobs = Job.all
+  end
+
+  # GET /jobs/1 or /jobs/1.json
+  def show
+    if user_signed_in? && (current_user.user_role.include? 'Employer')
+      #@jobs = Job.where(user_id: current_user.id)
```
Comment (Sharee-ongraph) [inline app/controllers/jobs_controller.rb:14]: remove commented code

File: app/views/jobs/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,63 @@
+<p id="notice"><%= notice %></p>
+
+
+<p>
+  <strong>Job ID:</strong>
+  <%= @job.id %>
```
Comment (Sharee-ongraph) [inline app/views/jobs/show.html.erb:6]: no need to show ID

File: app/views/static_pages/_employerview.html.erb [hunk 4]
```diff
@@ -0,0 +1,29 @@
+    <h1>Jobs Created</h1>
+<%# @jobs = Job.where(user_id:current_user.id)  %> <!-- for notes purpose -->
```
Comment (Sharee-ongraph) [inline app/views/static_pages/_employerview.html.erb:2]: remove commented code

**Review by mitaligaur — APPROVED** [review]

File: app/controllers/apply_jobs_controller.rb [hunk 5]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal:true
+
```
Comment (Sharee-ongraph) [inline app/controllers/apply_jobs_controller.rb:2]: add comment for class

File: app/controllers/apply_jobs_controller.rb [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+class ApplyJobsController < ApplicationController
+ 
+  def create
+    if user_signed_in? && (current_user.user_role.include? 'Job Seeker')
+      @apply_job = ApplyJob.create(user_id: params[:user_id].to_i, job_id: params[:job_id].to_i)
+      if @apply_job.valid?
+        @apply_job.save
+        job=Job.find(params[:job_id].to_i)
```
Comment (Sharee-ongraph) [inline app/controllers/apply_jobs_controller.rb:10]: create private method and set before filter for create and destroy method

File: app/controllers/apply_jobs_controller.rb [hunk 7]
```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal:true
+
+class ApplyJobsController < ApplicationController
+ 
+  def create
+    if user_signed_in? && (current_user.user_role.include? 'Job Seeker')
+      @apply_job = ApplyJob.create(user_id: params[:user_id].to_i, job_id: params[:job_id].to_i)
```
Comment (Sharee-ongraph) [inline app/controllers/apply_jobs_controller.rb:7]: keep it in separate method

File: app/controllers/jobs_controller.rb (hunk 2 above)
Comment (Sharee-ongraph) [inline app/controllers/jobs_controller.rb:14]: remove commented code

File: app/controllers/jobs_controller.rb [hunk 8]
```diff
@@ -0,0 +1,79 @@
+#frozen_string_literal:true
+
```
Comment (Sharee-ongraph) [inline app/controllers/jobs_controller.rb:2]: add class comments

File: app/controllers/static_pages_controller.rb [hunk 9]
```diff
@@ -2,7 +2,15 @@
```
Comment (Sharee-ongraph) [inline app/controllers/static_pages_controller.rb:2]: add frozen literal

File: app/controllers/static_pages_controller.rb [hunk 10]
```diff
@@ -2,7 +2,15 @@
 
 # class for static pages
 class StaticPagesController < ApplicationController
-  def home; end
+  def home
```
Comment (Sharee-ongraph) [inline app/controllers/static_pages_controller.rb:5]: add method comments

File: app/helpers/apply_job_helper.rb [hunk 11]
```diff
@@ -0,0 +1,11 @@
+module ApplyJobHelper
+    def applied?(user_id)
```
Comment (Sharee-ongraph) [inline app/helpers/apply_job_helper.rb:2]: add method comment

File: app/helpers/apply_job_helper.rb [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+        @apply_job = ApplyJob.find_by(user_id:user_id,job_id:params[:id].to_i)
+        if @apply_job.nil?
```
Comment (Sharee-ongraph) [inline app/helpers/apply_job_helper.rb:4]: use terniary operator

File: app/controllers/apply_jobs_controller.rb (hunk 1 above)
Comment (Eukst) [reply app/controllers/apply_jobs_controller.rb]: done

File: app/controllers/jobs_controller.rb (hunk 2 above)
Comment (Eukst) [reply app/controllers/jobs_controller.rb:14]: done

File: app/views/jobs/show.html.erb (hunk 3 above)
Comment (Eukst) [reply app/views/jobs/show.html.erb:6]: done

File: app/views/static_pages/_employerview.html.erb (hunk 4 above)
Comment (Eukst) [reply app/views/static_pages/_employerview.html.erb:2]: done

File: app/controllers/apply_jobs_controller.rb (hunk 5 above)
Comment (Eukst) [reply app/controllers/apply_jobs_controller.rb:2]: done

File: app/controllers/apply_jobs_controller.rb (hunk 6 above)
Comment (Eukst) [reply app/controllers/apply_jobs_controller.rb:10]: done

File: app/controllers/apply_jobs_controller.rb (hunk 7 above)
Comment (Eukst) [reply app/controllers/apply_jobs_controller.rb:7]: done

File: app/controllers/jobs_controller.rb (hunk 2 above)
Comment (Eukst) [reply app/controllers/jobs_controller.rb:14]: done

File: app/controllers/jobs_controller.rb (hunk 8 above)
Comment (Eukst) [reply app/controllers/jobs_controller.rb:2]: done

File: app/controllers/static_pages_controller.rb (hunk 9 above)
Comment (Eukst) [reply app/controllers/static_pages_controller.rb:2]: done

File: app/controllers/static_pages_controller.rb (hunk 10 above)
Comment (Eukst) [reply app/controllers/static_pages_controller.rb:5]: done

File: app/helpers/apply_job_helper.rb (hunk 11 above)
Comment (Eukst) [reply app/helpers/apply_job_helper.rb:2]: done

File: app/helpers/apply_job_helper.rb (hunk 12 above)
Comment (Eukst) [reply app/helpers/apply_job_helper.rb:4]: done

**Pull request merged**